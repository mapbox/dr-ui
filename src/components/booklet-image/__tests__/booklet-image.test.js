import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './booklet-image-test-cases.js';

describe('booklet-image', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.bigger.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.bigger;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
