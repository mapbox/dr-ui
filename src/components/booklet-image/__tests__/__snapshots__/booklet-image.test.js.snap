// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`booklet-image Basic renders as expected 1`] = `
<svg
  height={60}
  viewBox="0 0 108 108"
  width={60}
>
  <circle
    cx="53"
    cy="55"
    fill="#83e5af"
    r="45"
  />
  <path
    d="M63.238 70h-23.37a5.887 5.887 0 0 0-5.752 4.842 6.137 6.137 0 0 0-.11.868A5.898 5.898 0 0 0 39.763 82H63.77A5.341 5.341 0 0 0 69 76.554V64a5.885 5.885 0 0 1-5.762 6z"
    fill="#30aa67"
  />
  <rect
    fill="#3dbc7d"
    height="39"
    rx="4.744"
    ry="4.744"
    width="34"
    x="39"
    y="38"
  />
  <path
    d="M63.238 70.257A5.792 5.792 0 0 0 69 64.436v-32.18A4.257 4.257 0 0 0 64.743 28H38.257A4.257 4.257 0 0 0 34 32.257V75h.107a5.918 5.918 0 0 1 5.818-4.743"
    fill="#4bd18b"
  />
  <rect
    fill="#3dbc7d"
    height="15"
    rx="2.136"
    ry="2.136"
    width="25"
    x="39"
    y="33"
  />
</svg>
`;

exports[`booklet-image Bigger icon renders as expected 1`] = `
<svg
  height={120}
  viewBox="0 0 108 108"
  width={120}
>
  <circle
    cx="53"
    cy="55"
    fill="#83e5af"
    r="45"
  />
  <path
    d="M63.238 70h-23.37a5.887 5.887 0 0 0-5.752 4.842 6.137 6.137 0 0 0-.11.868A5.898 5.898 0 0 0 39.763 82H63.77A5.341 5.341 0 0 0 69 76.554V64a5.885 5.885 0 0 1-5.762 6z"
    fill="#30aa67"
  />
  <rect
    fill="#3dbc7d"
    height="39"
    rx="4.744"
    ry="4.744"
    width="34"
    x="39"
    y="38"
  />
  <path
    d="M63.238 70.257A5.792 5.792 0 0 0 69 64.436v-32.18A4.257 4.257 0 0 0 64.743 28H38.257A4.257 4.257 0 0 0 34 32.257V75h.107a5.918 5.918 0 0 1 5.818-4.743"
    fill="#4bd18b"
  />
  <rect
    fill="#3dbc7d"
    height="15"
    rx="2.136"
    ry="2.136"
    width="25"
    x="39"
    y="33"
  />
</svg>
`;
